windows/drivers/uwb/simulator/UwbSimulatorDevice.cxx

```diff
@@ -209,11 +172,15 @@ UwbSimulatorDevice::OnFileCreate(WDFDEVICE device, WDFREQUEST request, WDFFILEOB
         TraceLoggingPointer(request, "Request"),
         TraceLoggingPointer(file, "File"));
 
-    auto uwbSimulatorFileBuffer = GetUwbSimulatorFile(file);
+    auto uwbSimulatorFileBuffer = GetUwbSimulatorDeviceFile(file);
     auto uwbSimulatorFile = new (uwbSimulatorFileBuffer) UwbSimulatorDeviceFile(file, this);
     auto uwbSimulatorFileStatus = uwbSimulatorFile->Initialize();
     if (uwbSimulatorFileStatus == STATUS_SUCCESS) {
         uwbSimulatorFile->RegisterHandler(m_ddiHandler);
+
+        std::unique_lock deviceFilesLockExclusive{ m_deviceFilesGate };
+        m_deviceFiles.push_back(uwbSimulatorFile);
+        DbgPrint("%p added file object %p\n", m_wdfDevice, file);
     } else {

```

Comment:
Might be useful to add a log line in the "else" block

Reply:
Good idea.